test/check-machines-networks

```diff
@@ -25,16 +25,7 @@ from machinesxmls import TEST_NETWORK2_XML, TEST_NETWORK3_XML, TEST_NETWORK4_XML
 
 
 def getNetworkDevice(m):
-    net_devices_str = m.execute("virsh iface-list")
-    net_devices_str = net_devices_str.split("\n", 2)[2]  # Remove first 2 lines of table header
-    if net_devices_str not in ["\n", "\r\n"]:
-        device = net_devices_str.split(' ', 2)[1]  # Get the name of device, ignoring spacing before device string
-    else:  # If $virsh-iface list did not return any device, check virsh nodedev-list net:noh
-        net_devices_str = m.execute("virsh nodedev-list net")
-        net_devices_str = net_devices_str.split("\n", 2)[0]
-        device = net_devices_str.split("_", 2)[1]  # Ignore prefix (example: net_enp0s31f6_8c_16_45_5f_77_34)
-
-    return device
+    return m.execute("ls /sys/class/net | grep -v ^lo | head -n1").strip()
```

Comment:
As this is in a VM I suppose it works, I thought this was JS code.

Reply:
The file system order in /sys/class/net is "whatever", but `ls` sorts ASCIIbetically by default. I don't know the sorting order of `virsh iface-list`, but I can't imagine any profound technical reason why it would prefer wifi over ethernet? This smells rather arbitrary?

But this is only for that integration test which assumes it's running in our test VMs. Of course I'm happy to change the grepping/sorting if it helps you with running this test directly on your machine.

Reply:
It prefers WiFi over ethernet as I have no ethernet connected but it is available. Basically proving your assumptions don't hold in a real setup

Reply:
I'm not sure how to interpret this. An interface does not need to be connected (and I guess in most cases it actually should *not* be) in order to assign it to a VM -- that VM itself will in most cases configure it by itself.

Also, this isn't a real setup anyway -- the old test code's approach of "just pick the first device that `virsh iface-list` shows" is also completely arbitrary.

We can change the `ls` to something more elaborate, like `ip` -- but it seems neither of us knows how `virsh iface-list` sorted its devices. Even though I think it's irrelevant, let me do some research what list ordering it uses.